File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        # Solution has O(n) time complexity, and O(n) space complexity.
        values = {} # Dictionary of values in nums to their corresponding index.
        
        for i, val in enumerate(nums):
            if target - val in values: # If the number required in dictionary, we have the answer.
                return [values[target - val], i]
            else: # Otherwise, make an entry for the current value.
                values[val] = i

File: 1-two-sum/1-two-sum.java
class Solution {
    
    public int[] twoSum(int[] nums, int target) {
        
        // Solution: O(n) time complexity, and O(n) space complexity.
        
        // Dictionary of values in nums to their corresponding index.
        HashMap<Integer, Integer> values = new HashMap<Integer, Integer>();
        
        for (int i = 0; i < nums.length; i++) {
            int val = nums[i];
                
            // If the number required in dictionary, we have the answer.
            if (values.containsKey(target - val)) { 
                return new int[]{values.get(target - val), i};
            } else { // Otherwise, make an entry for the current value.
                values.put(val, i);
            }
            
        }
        
        return new int[2];
                    
    }
    
}